presto-main/src/main/java/com/facebook/presto/statusDetector/StatusDetector.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.statusDetector;
+
+import java.util.List;
+
+public interface StatusDetector
+{
+    List<String> getDisabled();
```

Comment:
Currently this class only allows for support of `disabled` or blocked nodes. We should allow for someone to also specify the allowed nodes - if any node other than the allowed list tries to connect then it should fail.

Reply:
Yes going to introduce that once this flow looks good.

Reply:
While I was initially thinking of getting this in Discovery service - seeing its complexity and this being the only use case - being here makes sense. 

I feel that this PR only should address this change. If we change the interface to `isAllowed(String host)` then we can just have the implementation take care of all the details - `return true` (for not care), `return false` (for disabled or if not in list) - just makes it a lot more flexible. What the proposed API prevents us from doing is to say give me a list of all the hosts that are disabled - but does it matter?

Reply:
added isAllowed instead.